### ISSUE TITLE
More explanation and or maybe a more thorough youtube example for embeddable 

### ISSUE THREAD

**Issue by DavidBates** [adobe/aem-core-wcm-components#1056]
## Feature Request

**Is your feature request related to a problem? Please describe.**
Having tried to follow along with all of the resources I could muster I could not in the end get a custom embeddable to work as advertised. I was able to finally get it to render the content I wanted but the attributes for my custom embeddable were rendering under the youtube embeddable in the dialog. Creating the embeddable was easy enough but the way it handles rendering the correct resources for the `cq:dialog` within the embeddable seems buggy or incomplete in documentation or sample. 

**Describe the solution you'd like**
I'd like to see the readme located at content/src/content/jcr_root/apps/core/wcm/components/embed/v1/embed be expanded in the area of creating your own embeddable especially the part about 

```To enable show/hide of the edit configuration options, the embeddable dialog snippet wrapper must have the following data attributes applied:

cmp-embed-dialog-edit-embeddableoptions="true"
cmp-embed-dialog-edit-showhidetargetvalue="<embeddableResourceType>"/>```

First, this seems like a typo with an extra /> on the end. (If this is a typo I'd be happy to submit a PR for it) Second, it is not clear where to add this to or what <embeddableResourceType> should be? Again if I can be pointed to the documentation around this or some other reference I'd be happy to submit a PR. 

**Are there alternatives?**
An alternative would be to have the youtube example embeddable contain these properties so we could simply copy and change the attributes. I almost filed this as a bug since if I copy the youtube embeddable to my own proxy component it will result in the options for youtube doubling up in one selection but be completely blank in another. 

**Other notes**
I tried with V 2.9.0 and V 3.0.0 due to #931

**Comment by msagolj** [comment]
@DavidBates, thx for the spotting the typo. About your other question:

_Second, it is not clear where to add this to or what should be?_

if ou follow the example link in the README there is a dialog definition which contains the entries you are looking for:
https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/embed/v1/embed/embeddable/youtube/_cq_dialog/.content.xml#L44

is this what you where looking for ?

**Comment by vladbailescu** [comment]
To sum up what one needs to do to create a configurable embeddable:

1. Create a _hidden_ component with a supertype of `core/wcm/components/embed/v1/embed/embeddable`.
2. Create a rendering HTL script suitable for what your want to render.
3. Create a `cq:dialog` node with only the configuration options needed for your embeddable.
4. Make sure to have the following data properties added to the container of the configuration options:
```
   cmp-embed-dialog-edit-embeddableoptions="true"
   cmp-embed-dialog-edit-showhidetargetvalue="<embeddableResourceType>"
```
where `<embeddableResourceType>` is the resource type of your custom embeeddable. See [YouTube embeddable options](https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/embed/v1/embed/embeddable/youtube/_cq_dialog/.content.xml#L42) for an example!
5. Make sure to namespace the JCR configuration options for your embeddable, to prevent clashes with other embeddables' options.

This is a rewording of https://github.com/adobe/aem-core-wcm-components/tree/master/content/src/content/jcr_root/apps/core/wcm/components/embed/v1/embed#custom-embeddable-fragment, do you find it easier to understand?

**Comment by DavidBates** [comment]
Yes, providing the link to that in the original instructions shows an example of what to use. I was purely using CRXDE to create my embeddable and it wasn't clear for the second option `cmp-embed-dialog-edit-showhidetargetvalue` that it was looking for the full path. I tried several things but of course, they were all wrong.

**Comment by DavidBates** [comment]
Know that I've had a chance to try this I see that these properties are on a granite:data node under the cq:dialog node. From the instructions, as a newer user, I don't know if I would know to create this node and add the properties there?

**Comment by msagolj** [comment]
Hi  @DavidBates , thx for providing a PR to make documentation more clear
